/* ========================= eCAL LICENSE =================================
 *
 * Copyright (C) 2016 - 2019 Continental Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * ========================= eCAL LICENSE =================================
*/

syntax = "proto3";

package IAA.Halo.SteeringWheel;

message Clear_Strip
{
  bool clr		= 1;
}
message SetColor_Strip
{
  uint32 red		= 1;
  uint32 green		= 2;
  uint32 blue		= 3;
  uint32 globalbrightness	= 4;   
}
message SetLength_Strip
{
  uint32 lenght 	= 1;
}
message Cmd_FadeIn
{
  uint32 red		= 1;
  uint32 green		= 2;
  uint32 blue		= 3;
  uint32 brightness	= 4;
  uint32 delay		= 5;
}
message Cmd_FadeOut
{
  uint32 red		= 1;
  uint32 green		= 2;
  uint32 blue		= 3;
  uint32 delay		= 4;
}
message HALO_SteeringWheel_Status
{
  uint32 alive_counter = 1;
}

message HALO_SteeringWheelCommand
{
  enum CommandList
  {
		NONE = 0;
    FADE_IN = 1;  // Beleuchtung der Lenkradtastatur einschalten/einblenden
    FADE_OUT = 2; // Beleuchtung der Lenkradtastatur ausschalten/ausblenden
    IDLE_OFF = 3; // Beleuchtung der Lenkradtastatur sofort ausschalten ohne Animation (hard off)
  }

  CommandList command = 1;
}
